Source/Scene/ModelInstanceCollection.js

```diff
@@ -0,0 +1,1006 @@
+     * @param {String} [options.basePath=''] The base path that paths in the glTF JSON are relative to.
+     * @param {Boolean} [options.dynamic=false] Hint if instance model matrices will be updated frequently.
+     * @param {Boolean} [options.show=true] Determines if the collection will be shown.
+     * @param {Boolean} [options.allowPicking=true] When <code>true</code>, each instance is pickable with {@link Scene#pick}.
+     * @param {Boolean} [options.asynchronous=true] Determines if model WebGL resource creation will be spread out over several frames or block until completion once all glTF files are loaded.
+     * @param {Boolean} [options.incrementallyLoadTextures=true] Determine if textures may continue to stream in after the model is loaded.
+     * @param {ShadowMode} [options.shadows=ShadowMode.ENABLED] Determines whether the collection casts or receives shadows from each light source.
+     * @param {Boolean} [options.debugShowBoundingVolume=false] For debugging only. Draws the bounding sphere for the collection.
+     * @param {Boolean} [options.debugWireframe=false] For debugging only. Draws the instances in wireframe.
+     *
+     * @exception {DeveloperError} Must specify either <options.gltf> or <options.url>, but not both.
+     * @exception {DeveloperError} Shader program cannot be optimized for instancing. Parameters cannot have any of the following semantics: MODEL, MODELINVERSE, MODELVIEWINVERSE, MODELVIEWPROJECTIONINVERSE, MODELINVERSETRANSPOSE.
+     *
+     * @private
+     */
+    function ModelInstanceCollection(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(options.gltf) && !defined(options.url)) {
+            throw new DeveloperError('Either options.gltf or options.url is required.');
+        }
+
+        if (defined(options.gltf) && defined(options.url)) {
+            throw new DeveloperError('Cannot pass in both options.gltf and options.url.');
+        }
+        //>>includeEnd('debug');
+
+        this.show = defaultValue(options.show, true);
+
+        this._instancingSupported = false;
+        this._dynamic = defaultValue(options.dynamic, false);
+        this._allowPicking = defaultValue(options.allowPicking, true);
+        this._cull = defaultValue(options.cull, true); // Undocumented option
+        this._ready = false;
+        this._readyPromise = when.defer();
+        this._state = LoadState.NEEDS_LOAD;
+        this._dirty = false;
+
+        this._instances = createInstances(this, options.instances);
+
+        // When the model instance collection is backed by an instanced 3d-tile,
+        // use its batch table resources to modify the shaders, attributes, and uniform maps.
+        this._batchTable = options.batchTable;
+
+        this._model = undefined;
+        this._vertexBufferData = undefined; // Hold onto the vertex buffer data when dynamic is true
```

Comment:
`_vertexBufferData` vs. `_vertexBuffer` does not have great semantics.  It's all data.  Is this CPU vs. GPU?  Perhaps an `ArrayBuffer` suffix?